datafusion/core/src/dataframe/mod.rs

```diff
@@ -156,7 +156,7 @@ impl Default for DataFrameWriteOptions {
 /// ```
 #[derive(Debug, Clone)]
 pub struct DataFrame {
-    session_state: SessionState,
+    session_state: Box<SessionState>,
```

Comment:
Maybe we can add a rationale here too so we don't forget why we did this. Perhaps something like

```suggestion
    // Box the (large) SessionState to reduce the size of DataFrame on the stack
    session_state: Box<SessionState>,
```

Reply:
done